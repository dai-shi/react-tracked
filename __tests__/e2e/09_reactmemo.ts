/* global page */

jest.setTimeout(15 * 1000);

describe('09_reactmemo', () => {
  const port = process.env.PORT || '8080';

  it('should work with recorded events', async () => {
    await page.goto(`http://localhost:${port}/`);

    await page.waitForSelector('body > #app > ul > li:nth-child(1) > input');
    await page.click('body > #app > ul > li:nth-child(1) > input');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(2) > input');
    await page.click('body > #app > ul > li:nth-child(2) > input');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(3) > input');
    await page.click('body > #app > ul > li:nth-child(3) > input');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(1) > input');
    await page.click('body > #app > ul > li:nth-child(1) > input');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(2) > input');
    await page.click('body > #app > ul > li:nth-child(2) > input');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(3) > input');
    await page.click('body > #app > ul > li:nth-child(3) > input');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();
  });
});
