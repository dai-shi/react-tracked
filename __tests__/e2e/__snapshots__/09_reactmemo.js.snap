// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`09_reactmemo should work with recorded events 1`] = `
"
    <div id=\\"app\\"><ul><li>numRendered: 8<input type=\\"checkbox\\"><span style=\\"text-decoration: line-through;\\">Wash dishes</span></li><li>numRendered: 4<input type=\\"checkbox\\"><span style=\\"text-decoration: none;\\">Study JS</span></li><li>numRendered: 6<input type=\\"checkbox\\"><span style=\\"text-decoration: none;\\">Buy ticket</span></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`09_reactmemo should work with recorded events 2`] = `
"
    <div id=\\"app\\"><ul><li>numRendered: 8<input type=\\"checkbox\\"><span style=\\"text-decoration: line-through;\\">Wash dishes</span></li><li>numRendered: 10<input type=\\"checkbox\\"><span style=\\"text-decoration: line-through;\\">Study JS</span></li><li>numRendered: 6<input type=\\"checkbox\\"><span style=\\"text-decoration: none;\\">Buy ticket</span></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`09_reactmemo should work with recorded events 3`] = `
"
    <div id=\\"app\\"><ul><li>numRendered: 8<input type=\\"checkbox\\"><span style=\\"text-decoration: line-through;\\">Wash dishes</span></li><li>numRendered: 10<input type=\\"checkbox\\"><span style=\\"text-decoration: line-through;\\">Study JS</span></li><li>numRendered: 12<input type=\\"checkbox\\"><span style=\\"text-decoration: line-through;\\">Buy ticket</span></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`09_reactmemo should work with recorded events 4`] = `
"
    <div id=\\"app\\"><ul><li>numRendered: 14<input type=\\"checkbox\\"><span style=\\"text-decoration: none;\\">Wash dishes</span></li><li>numRendered: 10<input type=\\"checkbox\\"><span style=\\"text-decoration: line-through;\\">Study JS</span></li><li>numRendered: 12<input type=\\"checkbox\\"><span style=\\"text-decoration: line-through;\\">Buy ticket</span></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`09_reactmemo should work with recorded events 5`] = `
"
    <div id=\\"app\\"><ul><li>numRendered: 14<input type=\\"checkbox\\"><span style=\\"text-decoration: none;\\">Wash dishes</span></li><li>numRendered: 16<input type=\\"checkbox\\"><span style=\\"text-decoration: none;\\">Study JS</span></li><li>numRendered: 12<input type=\\"checkbox\\"><span style=\\"text-decoration: line-through;\\">Buy ticket</span></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`09_reactmemo should work with recorded events 6`] = `
"
    <div id=\\"app\\"><ul><li>numRendered: 14<input type=\\"checkbox\\"><span style=\\"text-decoration: none;\\">Wash dishes</span></li><li>numRendered: 16<input type=\\"checkbox\\"><span style=\\"text-decoration: none;\\">Study JS</span></li><li>numRendered: 18<input type=\\"checkbox\\"><span style=\\"text-decoration: none;\\">Buy ticket</span></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;
